crates/lang/src/env_access.rs

```diff
@@ -111,6 +111,23 @@ where
 {
     /// Returns the address of the caller of the executed contract.
     ///
+    /// # Example
+    ///
+    /// ```
+    /// use ink_prelude;
+    /// # use core::convert::TryFrom;
+    /// # <ink_lang_ir::ItemImpl as TryFrom<syn::ItemImpl>>::try_from(syn::parse_quote! {
+    /// # impl MyContract {
+    ///  #[ink(message)]
+    ///  pub fn call_me(&self) {
+    ///     let caller = self.env().caller();
+    ///     let message = ink_prelude::format!("got a call from {:?}", caller);
+    ///     ink_env::debug_println(&message);
+    ///  }
+    /// # }
+    /// # }).unwrap();
```

Comment:
While this trick is used somewhere else to test parts of `ink_lang_ir` I would not recommend doing this and instead go with the full blown approach of setting up an ink! smart contract. The problem with this code is that you don't really test anything ink! related but only the syntax analysis which does not cover a lot of the underlying details of the code.

So even though the more optimal approach of setting up an entire ink! smart contract for every single doc test is daunting and will probably be a huge mess without proper library components (which we might want here) the current approach does not really suffice our needs of making sure that our doc tests are going to be battle tested properly.